src/Symfony/Component/Routing/Loader/Psr4DirectoryLoader.php

```diff
@@ -47,12 +51,33 @@ public function supports(mixed $resource, string $type = null): bool
         return ('attribute' === $type || 'annotation' === $type) && \is_array($resource) && isset($resource['path'], $resource['namespace']);
     }
 
+    public function forDirectory(string $currentDirectory): static
+    {
+        $loader = clone $this;
+        $loader->currentDirectory = $currentDirectory;
+
+        return $loader;
+    }
+
     private function loadFromDirectory(string $directory, string $psr4Prefix): RouteCollection
     {
         $collection = new RouteCollection();
+        $collection->addResource(new DirectoryResource($directory, '/\.php$/'));
+        $files = iterator_to_array(new \RecursiveIteratorIterator(
```

Comment:
I've copy/pasted the logic from the annotation directory loader to keep the same file order. If not, the routes won't be registered in the same order and that could be a BC break.
I haven't shared the logic between the classes as I supposed that we will deprecate the annotation loader soon enough.